xbmc/filesystem/File.cpp

```diff
@@ -765,6 +765,68 @@ int64_t CFile::GetPosition() const
   return -1;
 }
 
+bool XFILE::CFile::ReadString(std::vector<char>& line)
+{
+  if (!m_pFile)
+    return false;
+
+  line.clear();
+
+  if (m_pBuffer)
+  {
+    using traits = CFileStreamBuffer::traits_type;
+    CFileStreamBuffer::int_type aByte = m_pBuffer->sgetc();
+
+    while (aByte != traits::eof())
+    {
+      aByte = m_pBuffer->sbumpc();
+
+      if (aByte == traits::eof())
+        break;
+
+      if (aByte == traits::to_int_type('\n'))
+      {
+        if (m_pBuffer->sgetc() == traits::to_int_type('\r'))
+          m_pBuffer->sbumpc();
+        break;
+      }
+
+      if (aByte == traits::to_int_type('\r'))
+      {
+        if (m_pBuffer->sgetc() == traits::to_int_type('\n'))
+          m_pBuffer->sbumpc();
+        break;
+      }
+
+      line.emplace_back(traits::to_char_type(aByte));
+    }
+
+    return true;
+  }
+
+  try
+  {
+    // Read by buffer chuncks until to EOL or EOF
+    do
+    {
+      char bufferLine[1025];
+      if (!m_pFile->ReadString(bufferLine, 1024)) // EOF or error
```

Comment:
```suggestion
      char bufferLine[1025];
      if (!m_pFile->ReadString(bufferLine, sizeof(bufferLine))) // EOF or error
```
If you only want to read 1023 bytes from the file make `bufferLine` 1024. Either way the extra unused byte is confusing.

Reply:
applied